SPEC.md

```diff
@@ -41,21 +46,16 @@ set but it is typical to try and limit node size to less than one megabyte.
 
 ## IPLD `dir`
 
-A directory object represents a directory.
-
-The key of the map is a filename and is a CBOR text string encoded in UTF-8.
-The value of the map is another CBOR map with the following standard fields:
+A directory object represents a directory and has the following paths.
 
   - `type`: String with value of `'dir'`.
-  - `data`: Object or Link. 
-    - `data` as Object: values must be links to `file` or `dir` nodes.
-    - `data` as Link: Link must resolve to a `hamt` root node.
+  - `data`: 
+    - Object with keys as file/directory names that are string encoded in UTF-8.
+    - Values must be links to `file` or `dir` nodes.
+    - Link *may* resolve to a `hamt` root node.
```

Comment:
Doesn't it have to resolve to a node of the form `{type: ..., data: ...}`?

Reply:
I'm struggling to find the right language, but this does **not** refer to the values in the keyspace, it refers to the case where the value of `data` is a `hamt` node. I'm going to move this to the base definition next to "object with keys" to try and make it clearer.

Reply:
Maybe we can just say: A directory is an object mapping filenames to unixfs nodes (file, directory, etc.). We can explain (at the top) that a "node" could be behind a pointer (CID).

Reply:
Isn't the value always either an object of the form `{type: ..., data: ...}` *or* a link to an object with that same form?